src/components/EmailNotifications/EmailNotificationsForm.js

```diff
@@ -45,209 +46,223 @@ function EmailNotificationsForm({
+                        color: ${theme.positive};
+                      `}
+                    />
+                  )
                 }
+                adornmentPosition="end"
+                type="email"
+                wide
+                onChange={handleEmailAddressChange}
+                placeholder="you@example.org"
+                onBlur={handleEmailAddressBlur}
+              />
+            </Field>
+
+            {!insideModal && (
+              <div
                 css={`
-                  width: 100%;
-                  margin-bottom: 0px;
+                  display: flex;
+                  align-items: flex-end;
+                  margin-left: ${2 * GU}px;
                 `}
               >
-                <TextInput
-                  value={emailAddress}
-                  adornment={
-                    emailInvalid ? (
-                      <IconCross
-                        css={`
-                          color: ${theme.negative};
-                        `}
-                      />
-                    ) : emailAddress.trim() ? (
-                      <IconCheck
-                        css={`
-                          color: ${theme.positive};
-                        `}
-                      />
-                    ) : (
-                      <IconCheck
-                        css={`
-                          opacity: 0;
-                        `}
-                      />
-                    )
-                  }
-                  adornmentPosition="end"
-                  type="email"
-                  wide
-                  onChange={handleEmailAddressChange}
-                  placeholder="email@address.com"
-                  onBlur={handleEmailAddressBlur}
-                />
-              </Field>
-
-              {!isModal && (
-                <div
-                  css={`
-                    display: flex;
-                    align-items: flex-end;
-                    margin-left: ${2 * GU}px;
-                  `}
-                >
-                  <Button
-                    mode="strong"
-                    disabled={emailInvalid || !termsAccepted}
-                    onClick={handleOnSubscribeToNotifications}
-                    size="medium"
-                  >
-                    {existingEmail ? 'Update' : 'Subscribe'}
-                  </Button>
-                </div>
-              )}
-            </div>
-
-            {emailInvalid && (
-              <div>
-                <p
-                  css={`
-                    color: ${theme.negative};
-                    ${textStyle('body4')};
-                    text-align: left;
-                    height: 0;
-                    margin-top: ${0.5 * GU}px;
-                  `}
+                <Button
+                  mode="strong"
+                  type="submit"
+                  disabled={emailInvalid || !termsAccepted}
+                  onClick={handleOnSubscribeToNotifications}
```

Comment:
I think the idea was to wrap these components around a `form`, so with `type="submit"` here, the form's `onSubmit()` would be called.

Reply:
yes, i didn't push the `<form>` because is triggering the checkbox select and unselect for some reason regrdless that i am calling on `<form onSubmit={handleOnSubscribeToNotifications}>`